diff --git a/qt5/PKGBUILD b/qt5/PKGBUILD
index 1342a77..0f25d81 100644
--- a/qt5/PKGBUILD
+++ b/qt5/PKGBUILD
@@ -30,7 +30,7 @@ pkgname=('qt5-3d-debug'
          'qt5-x11extras-debug'
          'qt5-xmlpatterns-debug')
 pkgver=5.5.1
-pkgrel=9
+pkgrel=9.1
 arch=('i686' 'x86_64')
 url='http://www.qt.io/developers'
 license=('GPL3' 'LGPL' 'FDL' 'custom')
@@ -44,17 +44,22 @@ makedepends=('libxcb' 'xcb-proto' 'xcb-util' 'xcb-util-image' 'xcb-util-wm' 'xcb
 groups=('qt' 'qt5')
 _pkgfqn="qt-everywhere-opensource-src-${pkgver}"
 source=("http://download.qt-project.org/official_releases/qt/${pkgver%.*}/${pkgver}/single/${_pkgfqn}.tar.xz"
-        'assistant.desktop' 'designer.desktop' 'linguist.desktop' 'qdbusviewer.desktop')
+        'assistant.desktop' 'designer.desktop' 'linguist.desktop' 'qdbusviewer.desktop'
+        'division-by-zero.patch::http://code.qt.io/cgit/qt/qtbase.git/patch/?id=79447068579ea93d616d840bb8cbbf8adb1ed6ec')
 md5sums=('c2a249e8795d45200d997d96860d0353'
          'b2897dd6a2967bccf8f10e397aafee55'
          '76e150b20178d2d43801f7367232e5f7'
          '188da8f4c87316e730ebf1c6217bf5a0'
-         '322b419b16c75d4de0ee7ad0a246caa1')
+         '322b419b16c75d4de0ee7ad0a246caa1'
+         '43ed20fa76dc49aa5736fa6de9ddca03')
 options=('debug' '!strip')
 
 prepare() {
   cd ${_pkgfqn}
 
+  # Fix division by zero when using a recent Xvfb (QTBUG-49322/FS#48195)
+  patch -d qtbase -p1 -i "${srcdir}/division-by-zero.patch"
+
   # Build qmake using Arch {C,LD}FLAGS
   # This also sets default {C,CXX,LD}FLAGS for projects built using qmake
   sed -i -e "s|^\(QMAKE_CFLAGS_RELEASE.*\)|\1 ${CFLAGS}|" \

